src/gpu/nvidia/cudnn_convolution_impl.hpp

```diff
@@ -523,32 +524,72 @@ struct cudnn_convolution_impl_fwd_t : public cudnn_convolution_impl_base_t {
             }
         }
 
-        if (fused) {
-            auto err = cudnnConvolutionBiasActivationForward(handle, &scale,
-                    descs[io::x], x, weights_desc, weights, conv_desc,
-                    fwd_alg_kind, scratchpad, scratchpad_size, &beta,
-                    descs[io::y], output, descs[io::bias], bias,
-                    conv_bias_eltwise ? eltwise_desc : activation_desc,
-                    descs[io::y], output);
-            // try to fallback into standalone convolution
-            if (err == CUDNN_STATUS_NOT_SUPPORTED) {
-                fused = false;
-            } else {
-                CUDNN_CHECK_V(err);
+        float *y_fp32_data = nullptr;
+        dim_t size_output = 1;
+        if (dst_scale && data_types[io::y] == CUDNN_DATA_INT8) {
+            // Compute the size output to pass as a parameter to the
+            // memory allocation call.
+            for (size_t i = 0; i < ndims[io::y]; i++)
+                size_output *= dims[io::y][i];
+            CUDA_EXECUTE_FUNC(cuMemAlloc, (CUdeviceptr *)&y_fp32_data,
```

Comment:
We did use `scratchpad` instead of runtime memory allocation. We needed to change some additional files to get things to work. Let me know if there are new suggestions. Thanks.